ordered_associative_task/student/include/impl.h

```diff
@@ -15,4 +17,13 @@
  *
  * @return 1-5
  */
-size_t convertMarkTo5PointSystem(size_t markIn100Point);
+
+size_t convertMarkTo5PointSystem(size_t markIn100Point){
+    map<pair<size_t, size_t>, size_t> GradeIntervals = {{{0, 29},1},{{30, 49},2},{{50, 69},3},{{70, 89},4},{{90, 100},5}};
+    for (const auto it : GradeIntervals){
```

Comment:
You are doing the linear search, that's ineffective
Try to solve it with mapping {{0, 1}, {30, 2}, {50, 3}, {70, 4}, {90, 5}};